org-roam.el

```diff
@@ -823,6 +853,301 @@ for Org-ref cite links."
+  (let* ((path (org-element-property :path element))
+         (filename (org-roam--get-file-from-title path)))
+    (when filename
+      (let* ((start (org-element-property :begin element))
+             (end (org-element-property :end element))
+             (contents-begin (org-element-property :contents-begin element))
+             (contents-end (org-element-property :contents-end element))
+             (desc (and contents-begin contents-end
+                        (buffer-substring-no-properties contents-begin contents-end)))
+             (link-desc (if desc
+                            desc
+                          (org-roam--format-link-title path))))
+        (setf (buffer-substring start end)
+              (concat (org-roam--format-link filename link-desc)
+                      (make-string (org-element-property :post-blank element) ?\s)))))))
+
+(defun org-roam--get-title-from-file (path)
+  "Return title/aliases for a filename PATH in the Org-roam database.
+This function differs from `org-roam-db--get-titles' because it will
+match against a filename without an absolute path."
+  (org-roam-db-query [:select [titles] :from titles :where (like file $r1) :limit 1]
+                           (format "%%/%s%%" path)))
+
+(defun org-roam--convert-file-to-roam-link (element)
+  "Convert a standard org file-link to a custom roam-link.
+ELEMENT is the `org-element-at-point'.
+Ignore file-links that are not in Org-roam database."
+  (let ((path (org-element-property :path element)))
+    (when (org-roam--org-roam-file-p path)
+      (let* ((start (org-element-property :begin element))
+             (end (org-element-property :end element))
+             (contents-begin (org-element-property :contents-begin element))
+             (contents-end (org-element-property :contents-end element))
+             (desc (and contents-begin contents-end
+                        (buffer-substring-no-properties contents-begin contents-end)))
+             (list-titles (caar (org-roam--get-title-from-file path)))
+             (-compare-fn #'cl-equalp)
+             (title (if (-contains? list-titles desc)
+                            desc
+                      (car list-titles)))
+             (link-desc (if (string= title desc)
+                            nil
+                          desc)))
+        (when title
+          (setf (buffer-substring start end)
+              (concat (org-link-make-string (concat "roam:" title) link-desc)
+                      (make-string (org-element-property :post-blank element) ?\s))))))))
+
+(defun org-roam-convert-roam-to-file-link ()
+  "Convert a custom roam-link to standard org file-link."
+  (interactive)
+  (let ((elem (org-element-context)))
+    (when (and (eq 'link (car elem))
+               (string= "roam" (org-element-property :type elem)))
+      (org-roam--convert-roam-to-file-link elem))))
+
+(defun org-roam-convert-file-to-roam-link ()
```

Comment:
I was thinking of this work as primarily a way to lazy create links, much like a true Wiki. When would someone want to change an instantiated link back into a roam link?

Reply:
For easy editing. It's likely a narrow use case, but there's I don't think prescribing others' usage is good practice when the functionality is so easy to include. If it can convert one way, it should be able to convert back.